sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/ExcludeRegionTests.java

```diff
@@ -208,19 +226,29 @@ private List<String> getPreferredRegionList(CosmosAsyncClient client) {
         return preferredRegionList;
     }
 
-    private CosmosDiagnostics performDocumentOperation(
+    private CosmosDiagnosticsContext performDocumentOperation(
         CosmosAsyncContainer cosmosAsyncContainer,
         OperationType operationType,
         TestItem createdItem,
-        List<String> excludeRegions) {
+        List<String> excludeRegions,
+        CosmosEndToEndOperationLatencyPolicyConfig cosmosEndToEndOperationLatencyPolicyConfig) {
+
         if (operationType == OperationType.Query) {
             CosmosQueryRequestOptions queryRequestOptions = new CosmosQueryRequestOptions();
+
+            queryRequestOptions.setCosmosEndToEndOperationLatencyPolicyConfig(cosmosEndToEndOperationLatencyPolicyConfig);
+
             String query = String.format("SELECT * from c where c.id = '%s'", createdItem.getId());
             queryRequestOptions.setExcludedRegions(excludeRegions);
             FeedResponse<TestItem> itemFeedResponse =
                 cosmosAsyncContainer.queryItems(query, queryRequestOptions, TestItem.class).byPage().blockFirst();
 
-            return itemFeedResponse.getCosmosDiagnostics();
+            assertThat(itemFeedResponse).isNotNull();
+            CosmosDiagnostics cosmosDiagnostics = itemFeedResponse.getCosmosDiagnostics();
+
+            assertThat(cosmosDiagnostics).isNotNull();
+
+            return cosmosDiagnostics.getDiagnosticsContext();
```

Comment:
cosmosDiagnostics.getDiagnosticsContext() evaluated to 2 prior the region exclusion change. The `QueryPlan` call when to "non-excluded" region.